### PR TITLE
Error handling

### PR DESCRIPTION
**Pull request by Nkanyan**

### CODE REVIEW

File: Saasprin/Data/AppacitiveCloud/APCompanyStorage.cs [hunk 1]
```diff
@@ -27,18 +28,26 @@ public class APCompanyStorage : ICompanyStorage
...
 
-            return new Company()
+            try
             {
-                Id = appacitiveCompany.Id,
-                Name = appacitiveCompany.Get<string>("name"),
-                UtcCreateDate = appacitiveCompany.CreatedAt,
-                UtcUpdateDate = appacitiveCompany.LastUpdatedAt
+                var appacitiveCompany = await APObjects.GetAsync("company", companyId.ToString());
+                return new Company()
```
Comment (spraveen1112) [inline Saasprin/Data/AppacitiveCloud/APCompanyStorage.cs]: This should be outside the try block.

File: Saasprin/Data/AppacitiveCloud/APCompanyStorage.cs [hunk 2]
```diff
@@ -27,18 +28,26 @@ public class APCompanyStorage : ICompanyStorage
...
+                    Id = appacitiveCompany.Id,
+                    Name = appacitiveCompany.Get<string>("name"),
+                    UtcCreateDate = appacitiveCompany.CreatedAt,
+                    UtcUpdateDate = appacitiveCompany.LastUpdatedAt
 
-            };
+                };
+
+            }
+            catch (Exception )
```
Comment (spraveen1112) [inline Saasprin/Data/AppacitiveCloud/APCompanyStorage.cs]: The `catch(Exception)` is not needed. You can simply say `catch`.

File: Saasprin/Data/AppacitiveCloud/APCompanyStorage.cs [hunk 3]
```diff
@@ -46,7 +55,7 @@ public class APCompanyStorage : ICompanyStorage
             var query = Query.Property("name").IsEqualTo(companyName);
             var companiesResult = await APObjects.FindAllAsync("company", query);
             if (companiesResult.TotalRecords == 0)
-                throw new ObjectNotFoundException();
+                throw new Saasprin.Model.ObjectNotFoundException("Record Not Found.");
```
Comment (sirius17) [inline Saasprin/Data/AppacitiveCloud/APCompanyStorage.cs]: Message should describe the problem. `No company found with the given name.`

File: Saasprin/Data/AppacitiveCloud/APUserStorage.cs [hunk 4]
```diff
@@ -41,23 +41,32 @@ public class APUserStorage : IUserStorage
...
+                        UtcCreateDate = result.LoggedInUser.CreatedAt,
+                        UtcUpdateDate = result.LoggedInUser.LastUpdatedAt,
+
+                    }
+                };
+            }
+            catch (Exception)
+            {
+
+                throw new Saasprin.Model.InvalidObjectException("Wrong Username Or Password.");
```
Comment (sirius17) [inline Saasprin/Data/AppacitiveCloud/APUserStorage.cs]: Login failures are not InvalidObjectExceptions. You should create a specific InvalidCredentialsException as it represents a completely different business error.

File: Saasprin/Data/AppacitiveCloud/APUserStorage.cs [hunk 5]
```diff
@@ -41,23 +41,32 @@ public class APUserStorage : IUserStorage
...
-            var result = await creds.AuthenticateAsync();
-
-            return new AuthenticationResult()
+            try
             {
-                Token = result.UserToken,
-                User = new User()
+                var result = await creds.AuthenticateAsync();
+
+                return new AuthenticationResult()
```
Comment (sirius17) [inline Saasprin/Data/AppacitiveCloud/APUserStorage.cs]: This code should be outside the try block.

**Pull request merged**